### PR TITLE
refactor: HMI Node 

### LINKED ISSUE #156: Chat interface has no access to some tools when running HMI without the "whoami" parameter.
**Describe the bug**
Running the HMI without robot description parameter:

```
streamlit run src/rai_hmi/rai_hmi/streamlit_hmi_node.py
```

has some limitations, ie. it can't fetch the camera output.

When specifying the robot description, for example

```
ros2 run rai_whoami rai_whoami_node --ros-args -p robot_description_package:="rosbot_xl_whoami"
streamlit run src/rai_hmi/rai_hmi/streamlit_hmi_node.py rosbot_xl_whoami
```

then everything is fine.

Is it expected behavior? If so, maybe some user feedback could be helpful. Also README should mention any limitations.

**To Reproduce**

1. Run demo simulation
2. Run `streamlit run src/rai_hmi/rai_hmi/streamlit_hmi_node.py`
3. Ask for camera output.
4. Camera output is not displayed, the error mentions that the `rclpy.Node object has no attribute 'get_raw_message_from_topic'`

**Expected behavior**
An image from the camera.

**Platform**

- OS:Ubuntu 22.04
- ROS 2 Humble

**Version**
e0cff74cf4a0275a3301a3061442d60797297b57

### PR DESCRIPTION
**Pull request by maciejmajek**
Merge after #149 
## Purpose

This PR introduces a `BaseHMINode` class to streamline the implementation of various Human-Machine Interface (HMI) modalities (e.g., text-to-text, speech-to-speech). The goal is to simplify the current HMI architecture, which has separate and complex solutions for different modalities.

## Proposed Changes

This PR:
- Adds the `BaseHMINode`. This base class provides core functionality for HMI nodes, including loading the system wide system prompt, querying a FAISS index, publishing processing statuses, and handling feedback requests.
- Implements abstract methods that must be overridden by subclasses to handle feedback requests.

## Issues
Related: #142 

## Testing
TODO

### CODE REVIEW

File: src/rai_hmi/rai_hmi/voice_hmi.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        local_history: List[BaseMessage] = [
+            SystemMessage(content=self.system_prompt),
+            HumanMessage(content=feedback_prompt),
+        ]
+        local_history.extend(self.history)
+        response = self.agent.invoke({"user_input": "", "chat_history": local_history})
+        output = response["output"]
+
+        self.processing = False
+        return output
```
Comment (maciejmajek) [inline src/rai_hmi/rai_hmi/voice_hmi.py]: This implementation seems like a dead end. The feedback mechanism needs a redo- possibly into an action.

Comment (maciejmajek) [reply src/rai_hmi/rai_hmi/voice_hmi.py]: Refactored into action

File: src/rai/rai/agents/state_based.py [hunk 2]
```diff
@@ -80,6 +81,25 @@ class Report(BaseModel):
...
+    with open("artifact_database.pkl", "rb") as file:
+        artifact_database = pickle.load(file)
+        if tool_call_id not in artifact_database:
+            artifact_database[tool_call_id] = artifacts
+        else:
+            artifact_database[tool_call_id].extend(artifacts)
+    with open("artifact_database.pkl", "wb") as file:
+        pickle.dump(artifact_database, file)
+
+
```
Comment (maciejmajek) [inline src/rai/rai/agents/state_based.py]: This is just a POC solution for a significant problem. Currently, llm vendors do not support tool messages with images. This is handled by splitting MultimodalToolMessages into pairs of (ToolMessage and MultimodalHumanMessage). This works fine for standard operations, however handling this is in streamlit is very hard.

Robust solution should be based on Redis or similar

**Comment by maciejmajek** [comment]
image

Streamlit GUI is working

**Review by rachwalk — APPROVED** [review]
LGTM

**Pull request merged**